### PR TITLE
semgrep: clarify license

### PR DESCRIPTION
**Pull request by thypon**

### CODE REVIEW

File: assets/semgrep_rules/generated/nonfree/security_noaudit_novuln.yaml [hunk 1]
```diff
@@ -1,5 +1,100 @@
 ---
 rules:
+- id: contrib.dlint.dlint-equivalent.insecure-xml-use
+  message: Insecure XML parsing functionality, prefer 'defusedxml'
```
Comment (fmarier) [inline assets/semgrep_rules/generated/nonfree/security_noaudit_novuln.yaml:4]: `defusedxml` is what I use everywhere too, but I vaguely recall reading something from @bcaller about him not liking it (or maybe I'm misremembering).

**Review by fmarier — APPROVED** [review]
I did not review all of the YAML files carefully, but the oss/nonfree splitting changes look good.

File: assets/semgrep_rules/generated/nonfree/security_noaudit_novuln.yaml (hunk 1 above)
Comment (bcaller) [reply assets/semgrep_rules/generated/nonfree/security_noaudit_novuln.yaml:4]: In the distant past I have written a bandit rule that simply required a parser attribute to be set on the vulnerable methods like `fromstring` and a rule to ensure any parser is created with `resolve_entities=False` and I think `load_dtd=False`. This was sufficient to make use of the relevant libraries safe without getting in the way.

All other methods could be used as-is and no additional library required and typing worked.

At that time the defusedxml library was a bit unmaintained, the functions were missing some attributes of the python libraries, no typing, and I think some other functionality of the xml libraries was missing. At one stage it was [abandoned](https://github.com/tiran/defusedxml/issues/27#issuecomment-422435344). I'm not sure of the current state, perhaps it's much better. Now, the [Python docs](https://docs.python.org/3/library/xml.html#the-defusedxml-package) say:

> Use of this package is recommended for any server code that parses untrusted XML data

So, I guess it's OK. But note that the safe defaults are already implemented in systems with updated Python and expat library. E.g. compare
[3.10](https://docs.python.org/3.10/library/xml.html#xml-vulnerabilities) and [3.5](https://docs.python.org/3.5/library/xml.html#xml-vulnerabilities). Should maybe verify this for ourselves.

This rule will flag the import of any XML libraries which I think is too noisy. If you want to just create XML, semgrep will yell at you to use defusedxml. However, defusedxml does not provide any benefit if you are generating XML, only parsing untrusted XML.

I've not really come to a conclusion here but that's what I was getting at.

Comment (thypon) [reply assets/semgrep_rules/generated/nonfree/security_noaudit_novuln.yaml:4]: This file is generated and unused anyway. Only vulns.yaml and audit.yaml are, at the current stage.

**Pull request merged**